File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int length = strlen(text);

    int letter_count = 0;
    int word_count = 1;
    int sentence_count = 0;

    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            word_count++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;

        }

        if (text[i] > 64 && text[i] < 123)
        {
            letter_count++;
        }
    }

    int index = round(((5.88 * letter_count - 29.6 * sentence_count) / word_count) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

punctuation = ['.', '!', '?']

text = get_string("Text: ").split()

words = len(text)
letters = 0
sentences = 0

for word in text:
    for char in word:
        if char.isalpha():
            letters += 1
    if word[-1] in punctuation:
        sentences += 1

grade = int(round(((5.88 * letters - 29.6 * sentences) / words) - 15.8))
if grade < 1:
    print("Before Grade 1")
elif grade > 15:
    print("Grade 16+")
else:
    print("Grade ", grade)
